code/datums/status_effects/debuffs/staggered.dm

```diff
@@ -0,0 +1,42 @@
+/datum/status_effect/staggered
+	id = "staggered"
+	tick_interval = 0.5 SECONDS
+	alert_type = null
+	remove_on_fullheal = TRUE
+
+/datum/status_effect/staggered/on_creation(mob/living/new_owner, duration = 10 SECONDS)
+	src.duration = duration
+	return ..()
+
+/datum/status_effect/staggered/on_apply()
+
+	//a very mild animation, but you can't stagger the dead.
+	if(owner.stat == DEAD)
+		owner.do_stagger_animation(duration / 10)
+		return FALSE
+
+	RegisterSignal(owner, COMSIG_LIVING_DEATH, PROC_REF(clear_staggered))
+	owner.add_movespeed_modifier(/datum/movespeed_modifier/staggered)
+	return TRUE
+
+/datum/status_effect/staggered/on_remove()
+	UnregisterSignal(owner, COMSIG_LIVING_DEATH)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/staggered)
+	// Resetting both X and Y positioning when we remove, even though we only animates on the X axis
+	owner.pixel_y = owner.base_pixel_y
```

Comment:
Still modifying y here although you don't seem to ever set it?

Reply:
guh I'm just silly, lemme fix that.